code/modules/vending/_vending.dm

```diff
@@ -1118,4 +1138,10 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	if(isitem(target))
 		var/obj/item/I = target
 		I.custom_price = price
-		to_chat(user, "<span class='notice'>You set the price of [I] to [price] cr.</span>")
+		var/has_component = I.GetComponent(/datum/component/storage)
+		if(has_component)
+			for(var/atom/A in I.contents)
+				A.custom_price = price
```

Comment:
what happens when an item exists in a vendor and you set a different price to a newly stacking item?